discord/ui/select.py

```diff
@@ -499,7 +499,7 @@ def append_option(self, option: SelectOption) -> None:
             The number of options exceeds 25.
         """
 
-        if len(self._underlying.options) > 25:
+        if len(self._underlying.options) >= 25:
```

Comment:
This is not right..? Discord allows 25 options so why do we raise a ValueError when we have 25?

Reply:
Because the current code path is adding another item...?

Reply:
@DA-344 @AbstractUmbra 

The `append_option` function executes before the option is added to `self._underlying.options`, so `>= 25` is correct. You can reproduce this issue using the bot code and test code provided below.

### Bot code

```py
import discord

bot = discord.Client(
    intents=discord.Intents.all()
)

@bot.event
async def on_message(message: discord.Message):
    if message.content == "!test":
        select = discord.ui.Select()
        for i in range(1, 25 + 1):
            select.add_option(label=f"Test {i}", value=str(i))

        # add 26th option
        select.add_option(label="26", value="26")

        view = discord.ui.View()
        view.add_item(select)

        await message.channel.send("Select Test", view=view)

bot.run("token")
```

### Test code

```py
@pytest.mark.asyncio
async def test_add_option():
    select = discord.ui.Select()

    for i in range(1, 25 + 1):
        select.add_option(label=str(i), value=str(i))

    with pytest.raises(ValueError):
        select.add_option(label="26", value="26")
  ```